hash.c

```diff
@@ -4749,8 +4749,8 @@ env_delete(VALUE name)
  * deletes the environment variable and returns its value (ignoring the block):
  *   ENV['foo'] = '0'
  *   ENV.delete('foo') { |name| fail 'ignored' } # => "0"
- * Raises TypeError if +name+ is not a String and cannot be coerced with \#to_str:
- *   ENV.delete(Object.new) # => TypeError raised
+ * Raises an exception if +name+ is invalid.
+ * See {Invalid Names and Values}[#class-ENV-label-Invalid-Names+and+Values].
```

Comment:
Going forward, I'd like for all the setters to link to "Invalid Names and Values" (currently at line 6283).  That way the detailed info is always available, but does not clutter up the text of the method descriptions.